controllers/openstackcluster_controller.go

```diff
@@ -397,11 +397,12 @@ func reconcileNetworkComponents(log logr.Logger, osProviderClient *gophercloud.P
 			handleUpdateOSCError(openStackCluster, errors.Errorf("failed to find only one network (result: %v): %v", networkList, err))
 			return errors.Errorf("failed to find only one network (result: %v): %v", networkList, err)
 		}
-		openStackCluster.Status.Network = &infrav1.Network{
-			ID:   networkList[0].ID,
-			Name: networkList[0].Name,
-			Tags: networkList[0].Tags,
+		if openStackCluster.Status.Network == nil {
+			openStackCluster.Status.Network = &infrav1.Network{}
```

Comment:
Any reason not writing the following?
openStackCluster.Status.Network = &infrav1.Network{
  ID:   networkList[0].ID,
  Name: networkList[0].Name,
  Tags: networkList[0].Tags,
}

Reply:
This would override `openStackCluster.Status.Network.Router`.

Reply:
@hidekazuna so you think about dropping the lines 403 to 405 which contain:
```go
		openStackCluster.Status.Network.ID = networkList[0].ID
		openStackCluster.Status.Network.Name = networkList[0].Name
		openStackCluster.Status.Network.Tags = networkList[0].Tags
```

My intention was to keep the previous behaviour which would update the ID/Name/Tags variables if there are changes (don't know if this could even happen). Not updating the fields may be a regression.

Reply:
@seanschneeweiss @chrischdi Thanks explaining in detail, I understand.